### DISCUSSION TITLE
Closing the top level so no additional keys could be added

### DISCUSSION THREAD

**Discussion by dave08** [cue-lang/cue#1315] [Q&A]
Is there a way to close the top level so no additional keys could be added?

```
// c1.cue
key1: "is something"

key2: "should not be allowed

// c2.cue
close({
  key1: string
})

// run
cue export c2.cue c1.cue -o out.json

// was expecting to get an error, since key2 isn't allowed... but it just adds it.
```

**Comment by mvdan** [comment]
You can use a closed definition and then embed it at the root level, like https://github.com/cue-lang/cue/issues/2214 shows. Note that that's broken right now with `cue vet`, but it should work.
